client/dashboard/app/router.tsx

```diff
@@ -143,6 +143,19 @@ const siteSettingsSubscriptionGiftingRoute = createRoute( {
 	)
 );
 
+const siteSettingsTransferSiteRoute = createRoute( {
+	getParentRoute: () => siteRoute,
+	path: 'settings/transfer-site',
+	loader: ( { params: { siteSlug } } ) =>
+		queryClient.ensureQueryData( siteSettingsQuery( siteSlug ) ),
```

Comment:
Should we add a `beforeLoad` config here to redirect the user to some `/unauthorized` page when he's not the owner.

Reply:
Oh, good point!

Reply:
throwing in the component can work, but protecting routes in `beforeLoad` seems like a common practice in tanstack router apps. We do it in some of our routes already. Not sure what's best.

Reply:
I'm not sure we can move the "is the user the site owner" to `beforeLoad`. It happens before `loader`, in which the data fetching starts to take place, so I'm not sure how we can get the current `site` there in `beforeLoad` 🤔

Reply:
You're right :), let's throw in the component then for now :) 

Looks like we should be able to do something like

```
  if (isOwner) {
    throw redirect({ to: '/unauthorized' });
  }
```

within the render functions.